ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+            click?: BreadcrumbsClickCallbackFunction;
+        }
+        interface BreadcrumbsButtonsFormatter {
+            (
+                breadcrumb: Array<number|null|Point|SeriesType|SVGAElement>,
+                breadcrumbs: Breadcrumbs
+            ): (string);
+        }
+        interface DrilldownOptions {
+            breadcrumbs?: BreadcrumbsOptions;
+        }
+        interface SeparatorOptions {
+            symbol?: string;
+            size?: number;
+        }
+        interface TreemapSeriesOptions {
+            breadcrumbs?: BreadcrumbsOptions;
+        }
+        interface ChartLike {
+            breadcrumbs?: Breadcrumbs;
+        }
+        interface LangOptions {
+            mainBreadcrumb?: string;
+        }
+        class Breadcrumbs {
+            public constructor(chart: Chart, userOptions: Highcharts.BreadcrumbsOptions, isTreemap: boolean);
+
+            public breadcrumbsGroup: SVGElement;
+            public breadcrumbsList: Array<Array<number|null|Point|SeriesType|SVGAElement>>
+            public chart: Chart;
+            public isDirty: boolean;
+            public isTreemap: boolean;
+            public level: number;
+            public options: BreadcrumbsOptions;
+
+            public alignGroup(): void;
+            public createList(e?: any): void;
+            public destroyGroup(this: Breadcrumbs): void;
+            public draw(): void;
+            public multipleDrillUp(numb: number): void;
+            public init(chart: Chart, userOptions: Highcharts.BreadcrumbsOptions, isTreemap: boolean): void;
+            public redraw(): void;
+            public resetList(this: Breadcrumbs): void
+            public renderButton(breadcrumb: Array<number|Point|SVGAElement>, posX: number, posY: number):
+            SVGElement|void;
+            public renderSeparator(posX: number, posY: number): SVGElement|undefined;
+            public update(userOptions: Highcharts.BreadcrumbsOptions, redrawFlag: boolean): void;
+            public updateSingleButton(this: Breadcrumbs): void;
+            public setMargins(this: Breadcrumbs): void;
+        }
+    }
+}
+
+// Add language support.
+extend(
+    defaultOptions.lang,
+    /**
+     * @optionparent lang
+     */
+    {
+        /**
+         * @since    next
+         * @product  highcharts
+         */
+        mainBreadcrumb: 'Main'
+    }
+);
+
+/**
+ * The Breadcrumbs class
+ *
+ * @class
+ * @name Highcharts.Breadcrumbs
+ *
+ * @param {Highcharts.Chart} chart
+ *        Chart object
+ * @param {Highcharts.Options} userOptions
+ *        User options
+ */
+class Breadcrumbs {
+    /**
+     * Options for breadcrumbs
+     *
+     * @since   next
+     * @product highcharts
+     */
+    public static defaultBreadcrumbsOptions = {
+        /**
+         * The default padding for each button and separator in each direction.
+         *
+         * @type      {number}
+         * @since     next
+         */
+        buttonPadding: 5,
```

Comment:
As far as I can see from the code, this is actually the space _between_ the buttons/separators. The padding of the buttons is hardcoded to 3 (see separate comment). Instead, this option should be called `buttonSpacing`, which would be consistent with the range selector option.